### DISCUSSION TITLE
VSCode on Save adding space after colon (:)

### DISCUSSION THREAD

**Discussion by user3470** [tailwindlabs/tailwindcss#6284] [Help]
I'm trying to do
```js 
@apply hover:tw-bg-primary;
```

On Save it updates to
```js
@apply hover: tw-bg-primary;
```

With a space in it, I can't get rid of it. Please help

**Comment by phojie** [comment]
any updates on this one ?

**Reply by gerrcass** [reply]
I was struggling with this for a while. This only happened to me in the app.css file where I put the Tailwind directives (inside @layer base declaration). I managed to fix it like this:

1) Add a .vscode/settings.json file.
2) Paste the following.

`{
    "css.lint.unknownAtRules": "ignore",
    "[css]": {
        "editor.defaultFormatter": "bradlc.vscode-tailwindcss"
    }
}`

Let me know if there is any better solution

**Reply by rnrnshn** [reply]
Thanks. This has solved the issues

**Reply by ErisonSales** [reply]
Thank you problem solved

**Reply by kenshinman** [reply]
I want to buy you a drink for this 😁 @gerrcass

**Reply by gerrcass** [reply]
haha, you're welcome😂

Note that this code also works:

`.vscode/settings.json`

```
"files.associations": {
  "*.css": "tailwindcss"
}
```

**Comment by atonse** [comment]
I found a "cleaner" solution in the official TailwindCSS VSCode Extension readme – to associate all css files as tailwindcss files. By just doing this, tailwind probably uses its internal formatter and my css files don't get messed up anymore.

Place this in your `.vscode/settings.json`

```js
"files.associations": {
  "*.css": "tailwindcss"
}
```

**Reply by DevYemi** [reply]
where do i insert that piece of code ?, in the tailwind config file ?

**Reply by DevYemi** [reply]
This is not working for me, at the moment i'm just saving the css file without formatting, on window its crtl + k + shift + s

**Reply by ajcollins** [reply]
Thank you, @atonse. :+1: 

@DevYemi this can be added to your `.vscode/settings.json` if you open that folder as a workspace directly. You can also add a file association in the settings editor (File > Preferences > Settings) under Text Editor > Files. Enter `*.css` for the item, and `tailwindcss` for the value.

**Reply by DevYemi** [reply]
Thank you very much Alex, I have been able to fix it with your help 🙏🏾

**Reply by atonse** [reply]
Thank you for the reply – I modified my original comment to clarify where to put it.

**Reply by gabeboisvert** [reply]
Worked for me as well!

**Reply by joffreyBerrier** [reply]
Doesn't work for me I'm using css in a `.vue` file, not a dedicated file .css
Any solution for .vue file ? My solution is to change md:to @screen md {} ...

**Reply by jamessoncardozo** [reply]
> Thank you, @atonse. 👍
> 
> @DevYemi this can be added to your `.vscode/settings.json` if you open that folder as a workspace directly. You can also add a file association in the settings editor (File > Preferences > Settings) under Text Editor > Files. Enter `*.css` for the item, and `tailwindcss` for the value.

Thanks for your help. This save my day.

**Reply by exort-apps** [reply]
This helped for the Tailwind project, but will it affect CSS files when not using Tailwind?

**Comment by wlkns** [comment]
For anyone using Vue with Volar and the included formatter, this issue is still present because it reverts back to https://github.com/microsoft/vscode-css-languageservice behaviour on .vue files.

Read more @ https://github.com/johnsoncodehk/volar/issues/1448

**Comment by WilliamYi74** [comment]
i also get this problem inside .vue style, have any solution?

**Comment by ankush981** [comment]
I solved it by changing the default CSS formatter from VSCode one to Prettier+. Looks like the built-in formatter isn't aware of Tailwind-style stuff. 🤷🏻‍♂️

**Reply by 3li7u** [reply]
it works for me, thanks for sharing this.

**Comment by uglyeoin** [comment]
I'm having the same problem but not with Tailwind.  I'm just using this selector .team-member `.uk-width-1-2@m:nth-of-type(2) {` does anyone know how to solve it for this specific rule?

**Reply by uglyeoin** [reply]
I found out that I needed to escape the @ .team-member `.uk-width-1-2\@m:nth-of-type(2)`
